app/src/main/java/com/randomappsinc/studentpicker/grouping/GroupMakingSettingsViewHolder.java

```diff
@@ -98,8 +100,16 @@ void afterGroupNumChanged() {
     // Given the number of names per group or number of groups, returns the corresponding number to use all the names
     private String getMatchingNumber(String inputtedNumber) {
         int newNumber = Integer.parseInt(inputtedNumber);
-        int offset = (int) Math.ceil((double) settings.getNameListSize() / newNumber);
-        return String.valueOf(offset);
+        double offset = (double) settings.getNameListSize() / newNumber;
+        checkGroupBalance(offset);
```

Comment:
In general, we want to do single responsibility design. So every module, class, and API (like this one) is only responsible for 1 thing. This makes the code clean, compact, and easy to understand. By having checkGroupBalance() fire within this method that seemingly only calculates a string given its name, this almost feels like a side effect and violates single responsibility design. I feel like we should fire this in the onAfterTextChanged() block for each of the EditTexts. It will lead to redundant calls, but this operation is cheap and code readability is more important.